### ISSUE TITLE
The Drawer should not overlay the app bar

### ISSUE THREAD

**Issue by Cedware** [flutter/flutter#5704]
The Drawer overlays the Appbar, that causes that the first item of the drawer is hidden under the statusbar.
I think there should be at leat an option to place the Drawer under the appbar.

**Comment by Hixie** [comment]
Can you attach a screen shot showing what you mean? (`flutter screenshot` can take a screenshot)

**Comment by HansMuller** [comment]
https://material.google.com/patterns/navigation-drawer.html#navigation-drawer-behavior

Material design includes this feature under "Navigation drawer clipped under the app bar". Flutter should support that.

An appbar's height can be flexible, see https://material.google.com/patterns/scrolling-techniques.html#scrolling-techniques-behavior (see the "Flexible space" section).

We could just provide an option that offsets the top of the drawer by the standard toolbar height or by an arbitrary amount: 0 by default and often kToolBarHeight.

**Comment by eseidelGoogle** [comment]
I don't think this is an option (according to that MD link) for mobile layouts?  Maybe for tablets?

**Comment by mattarau** [comment]
Apparently, the links from @HansMuller don't work anymore.

What I think it's important to be considered here is that there are three types of Drawer: [Standard](https://material.io/design/components/navigation-drawer.html#standard-drawer), [Modal](https://material.io/design/components/navigation-drawer.html#modal-drawer) and [Bottom](https://material.io/design/components/navigation-drawer.html#bottom-drawer).

The implementation of the Drawer Component for flutter seems to only provide the `Modal` option since it is the one to be used in small screen devices but, for tablets and desktops, the recommendation is to use the `Standard` option. From the docs:

> Standard navigation drawers allow interaction with both screen content and the drawer at the same time. They can be used on tablet and desktop, but they aren’t suitable for mobile due to limited screen size.

> In a responsive layout grid, at a defined minimum breakpoint of at least 600dp width, a standard drawer should be replaced with a modal drawer.

**Comment by HansMuller** [comment]
The links to @HansMuller still work although sometimes the target drops the ball :-)

**Comment by hassan-2040** [comment]
Whats the latest news on this? Anyone?

**Comment by ShafiMunshi** [comment]
Wrap the drawer widget by SafeArea solves my problem.

**Comment by Piinks** [comment]
This request is being closed as it is now largely resolved or obsolete. Modern Flutter and Material 3 provide flexible navigation components like `NavigationDrawer` that handle varied layout requirements. Additionally, as noted in the issue comments, developers can achieve the "clipped" drawer effect by utilizing `SafeArea` or by configuring the `Scaffold` and `AppBar` to suit their specific layout needs without requiring new specialized properties in the legacy `Drawer` widget.

If there is more work to do here, please let us know by filing a new issue with up to date information. Thanks!
